### ISSUE TITLE
pandas issue

### ISSUE THREAD

**Issue by mkardish** [MrOlm/drep#189]
Hi! Connecting an issue caught using the atlas pipeline with a pandas conflict. Setting pandas version to 1.5.1 instead of 2.0.0 allows dRep to finish the run. Thanks SilasK.

https://github.com/metagenome-atlas/atlas/issues/625

Here was the error.
```
***************************************************
    ..:: dRep dereplicate Step 1. Filter ::..
***************************************************
    
Will filter the genome list
Loading genomes from a list
325 genomes were input to dRep
Calculating genome info of genomes
100.00% of genomes passed length filtering
100.00% of genomes passed checkM filtering
***************************************************
    ..:: dRep dereplicate Step 2. Cluster ::..
***************************************************
    
Running primary clustering
Running pair-wise MASH clustering
Traceback (most recent call last):
  File "/p/work1/mkardish/subs/databases/conda_envs/e4ca0a910149c0c7b21c70f20a241e3d_/bin/dRep", line 32, in <module>
    Controller().parseArguments(args)
  File "/p/work1/mkardish/subs/databases/conda_envs/e4ca0a910149c0c7b21c70f20a241e3d_/lib/python3.10/site-packages/drep/controller.py", line 100, in parseArguments
    self.dereplicate_operation(**vars(args))
  File "/p/work1/mkardish/subs/databases/conda_envs/e4ca0a910149c0c7b21c70f20a241e3d_/lib/python3.10/site-packages/drep/controller.py", line 48, in dereplicate_operation
    drep.d_workflows.dereplicate_wrapper(kwargs['work_directory'],**kwargs)
  File "/p/work1/mkardish/subs/databases/conda_envs/e4ca0a910149c0c7b21c70f20a241e3d_/lib/python3.10/site-packages/drep/d_workflows.py", line 37, in dereplicate_wrapper
    drep.d_cluster.controller.d_cluster_wrapper(wd, **kwargs)
  File "/p/work1/mkardish/subs/databases/conda_envs/e4ca0a910149c0c7b21c70f20a241e3d_/lib/python3.10/site-packages/drep/d_cluster/controller.py", line 179, in d_cluster_wrapper
    GenomeClusterController(workDirectory, **kwargs).main()
  File "/p/work1/mkardish/subs/databases/conda_envs/e4ca0a910149c0c7b21c70f20a241e3d_/lib/python3.10/site-packages/drep/d_cluster/controller.py", line 32, in main
    self.run_primary_clustering()
  File "/p/work1/mkardish/subs/databases/conda_envs/e4ca0a910149c0c7b21c70f20a241e3d_/lib/python3.10/site-packages/drep/d_cluster/controller.py", line 100, in run_primary_clustering
    Mdb, Cdb, cluster_ret = drep.d_cluster.compare_utils.all_vs_all_MASH(self.Bdb, self.wd.get_dir('MASH'), **self.kwargs)
  File "/p/work1/mkardish/subs/databases/conda_envs/e4ca0a910149c0c7b21c70f20a241e3d_/lib/python3.10/site-packages/drep/d_cluster/compare_utils.py", line 115, in all_vs_all_MASH
    Cdb, cluster_ret = cluster_mash_database(Mdb, **kwargs)
  File "/p/work1/mkardish/subs/databases/conda_envs/e4ca0a910149c0c7b21c70f20a241e3d_/lib/python3.10/site-packages/drep/d_cluster/compare_utils.py", line 279, in cluster_mash_database
    linkage_db = db.pivot("genome1","genome2","dist")
TypeError: DataFrame.pivot() takes 1 positional argument but 4 were given
```

**Comment by nick-youngblut** [comment]
I'm getting the same issue:

```
Running primary clustering
Running pair-wise MASH clustering
Traceback (most recent call last):
  File "/opt/conda/envs/drep/bin/dRep", line 32, in <module>
    Controller().parseArguments(args)
  File "/opt/conda/envs/drep/lib/python3.11/site-packages/drep/controller.py", line 100, in parseArguments
    self.dereplicate_operation(**vars(args))
  File "/opt/conda/envs/drep/lib/python3.11/site-packages/drep/controller.py", line 48, in dereplicate_operation
    drep.d_workflows.dereplicate_wrapper(kwargs['work_directory'],**kwargs)
  File "/opt/conda/envs/drep/lib/python3.11/site-packages/drep/d_workflows.py", line 37, in dereplicate_wrapper
    drep.d_cluster.controller.d_cluster_wrapper(wd, **kwargs)
  File "/opt/conda/envs/drep/lib/python3.11/site-packages/drep/d_cluster/controller.py", line 179, in d_cluster_wrapper
    GenomeClusterController(workDirectory, **kwargs).main()
  File "/opt/conda/envs/drep/lib/python3.11/site-packages/drep/d_cluster/controller.py", line 32, in main
    self.run_primary_clustering()
  File "/opt/conda/envs/drep/lib/python3.11/site-packages/drep/d_cluster/controller.py", line 100, in run_primary_clustering
    Mdb, Cdb, cluster_ret = drep.d_cluster.compare_utils.all_vs_all_MASH(self.Bdb, self.wd.get_dir('MASH'), **self.kwargs)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/drep/lib/python3.11/site-packages/drep/d_cluster/compare_utils.py", line 115, in all_vs_all_MASH
    Cdb, cluster_ret = cluster_mash_database(Mdb, **kwargs)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/drep/lib/python3.11/site-packages/drep/d_cluster/compare_utils.py", line 279, in cluster_mash_database
    linkage_db = db.pivot("genome1","genome2","dist")
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

I'm using v3.4.2

**Comment by MrOlm** [comment]
Hi both-

Thanks for the bug report. This is indeed a pandas problem.

I'll work on pushing an update as soon as possible, but for now downgrading dRep to `1.5.1` should fix it

-Matt

**Comment by MrOlm** [comment]
Fixed in dRep version 3.4.3, pushed today
